### ISSUE TITLE
Microsoft Edge checkout can not be completed

### ISSUE THREAD

**Issue by MikiAlemayehu** [paypal/paypal-checkout#487]
### Steps to reproduce
1. Start to checkout from Windows edge on in-context checkout.
2. User try to login and press pay
3. Checkout fails

### Affected browsers

- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [X ] Edge
- [ ] IE
- [ ] Chrome Mobile/Tablet
- [ ] Safari Mobile/Tablet
- [ ] Web View / Safari ViewController
- [ ] Other

Additional source code and information. 

![windwos edge 1](https://user-images.githubusercontent.com/30797581/31162024-fb5bc3e2-a915-11e7-97f4-3eb2469b2948.png)

![windows edge 2](https://user-images.githubusercontent.com/30797581/31162023-fb535964-a915-11e7-8ab5-901122bb3d43.png)

![edge](https://user-images.githubusercontent.com/30797581/31162025-fb5c562c-a915-11e7-8f99-e0122282b8ac.png)

[post-robot] fullpage :: jo-pp-node-demo.herokuapp.com/vault Error: postrobot_message_ack xcomponent_check_close error:
  - postrobot_post_message: Error: Permission denied
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2949:17)
   at Array.prototype.forEach (native code)
   at SEND_MESSAGE_STRATEGIES[] (https://www.paypalobjects.com/api/checkout.js:2948:13)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2902:25)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8189:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2900:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8172:25)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8189:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8171:21)
   at ZalgoPromise.map (https://www.paypalobjects.com/api/checkout.js:8170:60)

  - postrobot_bridge: Error: Can only use bridge to communicate between two different windows, not between frames
   at SEND_MESSAGE_STRATEGIES[] (https://www.paypalobjects.com/api/checkout.js:2955:101)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2902:25)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8189:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2900:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8172:25)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8189:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8171:21)
   at ZalgoPromise.map (https://www.paypalobjects.com/api/checkout.js:8170:60)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2899:17)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8189:21)

  - postrobot_global: Error: Post message through global disabled between different domain windows
   at SEND_MESSAGE_STRATEGIES[] (https://www.paypalobjects.com/api/checkout.js:2959:87)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2902:25)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8189:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2900:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8172:25)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8189:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8171:21)
   at ZalgoPromise.map (https://www.paypalobjects.com/api/checkout.js:8170:60)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2899:17)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8189:21)

   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2913:35)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8065:33)
   at ZalgoPromise.prototype.dispatch (https://www.paypalobjects.com/api/checkout.js:8062:26)
   at ZalgoPromise.prototype.then (https://www.paypalobjects.com/api/checkout.js:8107:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2899:17)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8189:21)
   at sendMessage (https://www.paypalobjects.com/api/checkout.js:2884:13)
   at respond (https://www.paypalobjects.com/api/checkout.js:2818:17)
   at _RECEIVE_MESSAGE_TYPE[] (https://www.paypalobjects.com/api/checkout.js:2829:13)
   at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2764:25)
checkout.js (3224,25)

myDeviceData{"correlation_id":"fa135cd0b4e04a555f7e8d941dfbb07a"}
vault (127,7)

HTML1300: Navigation occurred.
checkoutVault
HTML1527: DOCTYPE expected. Consider adding a valid HTML5 doctype: "<!DOCTYPE html>".
checkoutVault (1,1)
DOM7011: The code on this page disabled back and forward caching. For more information, see: http://go.microsoft.com/fwlink/?LinkID=291337
vault
SCRIPT7002: XMLHttpRequest: Network Error 0x80070005, Access is denied.

vault

SCRIPT5007: Unable to get property 'length' of undefined or null reference
hermes (7118,13)

HTML1300: Navigation occurred.
vault
HTML1527: DOCTYPE expected. Consider adding a valid HTML5 doctype: "<!DOCTYPE html>".
vault (1,1)
ie_meta_compatibility_header_missing [object Object]
checkout.js (327,80)
   "ie_meta_compatibility_header_missing"
   {
      [functions]: ,
      __proto__: { },
      message: "Drop tag: <meta http-equiv="X-UA-Compatible" content="IE=edge">",
      pageID: "46fc5e934e",
      timestamp: 1507095301605,
      windowID: "8b2e8e4d40"
   }

myDeviceData{"correlation_id":"e03cc939d7f9a4e18d6311d11e444c82"}
vault (127,7)

**Comment by bluepnume** [comment]
A few questions: 

- Does this happen every time?
- Can you share the exact Edge version?
- Can you share the url for the page you're testing on?

**Comment by bluepnume** [comment]
@MikiAlemayehu still hoping for this info, thanks!

**Comment by MikiAlemayehu** [comment]
@bluepnume 
Hi Daniel,
I am sorry for a late reply. I was on the business trip and couldn't access to internet.

1. It does happen once in 10-20 times.
2. Edge version
Microsoft Edge 38.14393.1066.0
Microsoft EdgeHTML 14.14393
3. URL (Sandobox)
http://jo-pp-node-demo.herokuapp.com/

**Comment by benzookapi** [comment]
Hello @bluepnume ,  I am  the origin reporter of this. the demo site above is created by me.
I am taking over @MikiAlemayehu  work.
What is the status of this?

**Comment by bluepnume** [comment]
Hi @benzookapi. So far I have been unable to repro with the details provided. Are you able to personally hit this error case?

**Comment by benzookapi** [comment]
@bluepnume Hi Daniel, sorry for late reply and thank you for fixing it.  I will check it later but I don't have my own Windows Edge environment so it will take time.

**Comment by benzookapi** [comment]
@bluepnume  Hi as Miki tested, the situation is the same and the fix seems not working..
Thanks.
![image001](https://user-images.githubusercontent.com/899580/32475609-130a8f36-c3b6-11e7-8de3-1902bc3bf8ff.png)

The browser console log:

ppxo_button_mutliple_click_debounce [object Object]
logger.js (65,13)
   "ppxo_button_mutliple_click_debounce"
   {
      [functions]: ,
      __proto__: { },
      country: "JP",
      env: "sandbox",
      host: "www.sandbox.paypal.com",
      lang: "ja",
      pageID: "97a4d300d5",
      path: "/webapps/hermes/button",
      referer: "https://jo-pp-node-demo.herokuapp.com",
      timestamp: 1509609481278,
      uid: "47c74f8417",
      ver: "4.0.156",
      windowID: "2c85ce08f2"
   }
 
ppxo_unhandled_error [object Object]
logger.js (65,13)
   "ppxo_unhandled_error"
   {
      [functions]: ,
      __proto__: { },
      country: "US",
      env: "production",
      errtype: "[object Error]",
      host: "jo-pp-node-demo.herokuapp.com",
      lang: "en",
      pageID: "030258b95a",
      path: "/",
      referer: "jo-pp-node-demo.herokuapp.com",
      stack: "Error: No window to focus
   at ParentComponent.prototype.focus (https://www.paypalobjects.com/api/checkout.js:5977:35)
   at options.overrides.focus (https://www.paypalobjects.com/api/checkout.js:5724:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3438:21)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3437:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2858:17)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at _RECEIVE_MESSAGE_TYPE[] (https://www.paypalobjects.com/api/checkout.js:2852:13)
   at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2785:25)
   at messageListener (https://www.paypalobjects.com/api/checkout.js:2806:13)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3384:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8202:37)
   at ZalgoPromise.prototype.dispatch (https://www.paypalobjects.com/api/checkout.js:8189:26)
   at ZalgoPromise.prototype.reject (https://www.paypalobjects.com/api/checkout.js:8176:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8214:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8202:37)
   at ZalgoPromise.prototype.dispatch (https://www.paypalobjects.com/api/checkout.js:8189:26)
   at ZalgoPromise.prototype.reject (https://www.paypalobjects.com/api/checkout.js:8176:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8142:42)
   at responseListener.respond (https://www.paypalobjects.com/api/checkout.js:3639:43)",
      timestamp: 1509609617228,
      uid: "47c74f8417",
      ver: "4.0.156",
      windowID: "b03040dba1"
   }
 
ppxo_unhandled_error [object Object]
logger.js (65,13)
   "ppxo_unhandled_error"
   {
      [functions]: ,
      __proto__: { },
      country: "US",
      env: "production",
      errtype: "[object Error]",
      host: "jo-pp-node-demo.herokuapp.com",
      lang: "en",
      pageID: "030258b95a",
      path: "/",
      referer: "jo-pp-node-demo.herokuapp.com",
      stack: "Error: No window to focus
   at ParentComponent.prototype.focus (https://www.paypalobjects.com/api/checkout.js:5977:35)
   at options.overrides.focus (https://www.paypalobjects.com/api/checkout.js:5724:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3438:21)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3437:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2858:17)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at _RECEIVE_MESSAGE_TYPE[] (https://www.paypalobjects.com/api/checkout.js:2852:13)
   at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2785:25)
   at messageListener (https://www.paypalobjects.com/api/checkout.js:2806:13)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3384:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8202:37)
   at ZalgoPromise.prototype.dispatch (https://www.paypalobjects.com/api/checkout.js:8189:26)
   at ZalgoPromise.prototype.reject (https://www.paypalobjects.com/api/checkout.js:8176:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8214:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8202:37)
   at ZalgoPromise.prototype.dispatch (https://www.paypalobjects.com/api/checkout.js:8189:26)
   at ZalgoPromise.prototype.reject (https://www.paypalobjects.com/api/checkout.js:8176:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8142:42)
   at responseListener.respond (https://www.paypalobjects.com/api/checkout.js:3639:43)",
      timestamp: 1509609617235,
      uid: "47c74f8417",
      ver: "4.0.156",
      windowID: "b03040dba1"
   }
 
ppxo_unhandled_error [object Object]
logger.js (65,13)
   "ppxo_unhandled_error"
   {
      [functions]: ,
      __proto__: { },
      country: "US",
      env: "production",
      errtype: "[object Error]",
      host: "jo-pp-node-demo.herokuapp.com",
      lang: "en",
      pageID: "030258b95a",
     path: "/",
      referer: "jo-pp-node-demo.herokuapp.com",
      stack: "Error: No window to focus
   at ParentComponent.prototype.focus (https://www.paypalobjects.com/api/checkout.js:5977:35)
   at options.overrides.focus (https://www.paypalobjects.com/api/checkout.js:5724:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3438:21)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3437:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2858:17)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at _RECEIVE_MESSAGE_TYPE[] (https://www.paypalobjects.com/api/checkout.js:2852:13)
   at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2785:25)
   at messageListener (https://www.paypalobjects.com/api/checkout.js:2806:13)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3384:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8202:37)
   at ZalgoPromise.prototype.dispatch (https://www.paypalobjects.com/api/checkout.js:8189:26)
   at ZalgoPromise.prototype.reject (https://www.paypalobjects.com/api/checkout.js:8176:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8214:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8202:37)
   at ZalgoPromise.prototype.dispatch (https://www.paypalobjects.com/api/checkout.js:8189:26)
   at ZalgoPromise.prototype.reject (https://www.paypalobjects.com/api/checkout.js:8176:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8142:42)
   at responseListener.respond (https://www.paypalobjects.com/api/checkout.js:3639:43)",
      timestamp: 1509609617238,
      uid: "47c74f8417",
      ver: "4.0.156",
      windowID: "b03040dba1"
   }
 
ppxo_unhandled_error [object Object]
logger.js (65,13)
   "ppxo_unhandled_error"
   {
      [functions]: ,
      __proto__: { },
      country: "US",
      env: "production",
      errtype: "[object Error]",
      host: "jo-pp-node-demo.herokuapp.com",
      lang: "en",
      pageID: "030258b95a",
      path: "/",
      referer: "jo-pp-node-demo.herokuapp.com",
      stack: "Error: No window to focus
   at ParentComponent.prototype.focus (https://www.paypalobjects.com/api/checkout.js:5977:35)
   at options.overrides.focus (https://www.paypalobjects.com/api/checkout.js:5724:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3438:21)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3437:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2858:17)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at _RECEIVE_MESSAGE_TYPE[] (https://www.paypalobjects.com/api/checkout.js:2852:13)
   at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2785:25)
   at messageListener (https://www.paypalobjects.com/api/checkout.js:2806:13)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3384:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8202:37)
   at ZalgoPromise.prototype.dispatch (https://www.paypalobjects.com/api/checkout.js:8189:26)
   at ZalgoPromise.prototype.reject (https://www.paypalobjects.com/api/checkout.js:8176:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8214:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8202:37)
   at ZalgoPromise.prototype.dispatch (https://www.paypalobjects.com/api/checkout.js:8189:26)
   at ZalgoPromise.prototype.reject (https://www.paypalobjects.com/api/checkout.js:8176:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8142:42)
   at responseListener.respond (https://www.paypalobjects.com/api/checkout.js:3639:43)",
      timestamp: 1509609617241,
      uid: "47c74f8417",
      ver: "4.0.156",
      windowID: "b03040dba1"
   }
 
ppxo_unhandled_error [object Object]
logger.js (65,13)
   "ppxo_unhandled_error"
   {
      [functions]: ,
      __proto__: { },
      country: "US",
      env: "production",
      errtype: "[object Error]",
      host: "jo-pp-node-demo.herokuapp.com",
      lang: "en",
      pageID: "030258b95a",
      path: "/",
      referer: "jo-pp-node-demo.herokuapp.com",
      stack: "Error: No window to focus
   at ParentComponent.prototype.focus (https://www.paypalobjects.com/api/checkout.js:5977:35)
   at options.overrides.focus (https://www.paypalobjects.com/api/checkout.js:5724:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3438:21)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3437:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2858:17)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at _RECEIVE_MESSAGE_TYPE[] (https://www.paypalobjects.com/api/checkout.js:2852:13)
   at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2785:25)
   at messageListener (https://www.paypalobjects.com/api/checkout.js:2806:13)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3384:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8202:37)
   at ZalgoPromise.prototype.dispatch (https://www.paypalobjects.com/api/checkout.js:8189:26)
   at ZalgoPromise.prototype.reject (https://www.paypalobjects.com/api/checkout.js:8176:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8214:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8202:37)
   at ZalgoPromise.prototype.dispatch (https://www.paypalobjects.com/api/checkout.js:8189:26)
   at ZalgoPromise.prototype.reject (https://www.paypalobjects.com/api/checkout.js:8176:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8142:42)
   at responseListener.respond (https://www.paypalobjects.com/api/checkout.js:3639:43)",
      timestamp: 1509609617244,
      uid: "47c74f8417",
      ver: "4.0.156",
      windowID: "b03040dba1"
   }
 
ppxo_unhandled_error [object Object]
logger.js (65,13)
   "ppxo_unhandled_error"
   {
      [functions]: ,
      __proto__: { },
      country: "US",
      env: "production",
      errtype: "[object Error]",
      host: "jo-pp-node-demo.herokuapp.com",
      lang: "en",
      pageID: "030258b95a",
      path: "/",
      referer: "jo-pp-node-demo.herokuapp.com",
      stack: "Error: No window to focus
   at ParentComponent.prototype.focus (https://www.paypalobjects.com/api/checkout.js:5977:35)
   at options.overrides.focus (https://www.paypalobjects.com/api/checkout.js:5724:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3438:21)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3437:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2858:17)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at _RECEIVE_MESSAGE_TYPE[] (https://www.paypalobjects.com/api/checkout.js:2852:13)
   at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2785:25)
   at messageListener (https://www.paypalobjects.com/api/checkout.js:2806:13)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3384:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8202:37)
   at ZalgoPromise.prototype.dispatch (https://www.paypalobjects.com/api/checkout.js:8189:26)
   at ZalgoPromise.prototype.reject (https://www.paypalobjects.com/api/checkout.js:8176:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8214:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8202:37)
   at ZalgoPromise.prototype.dispatch (https://www.paypalobjects.com/api/checkout.js:8189:26)
   at ZalgoPromise.prototype.reject (https://www.paypalobjects.com/api/checkout.js:8176:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8142:42)
   at responseListener.respond (https://www.paypalobjects.com/api/checkout.js:3639:43)",
      timestamp: 1509609617247,
      uid: "47c74f8417",
      ver: "4.0.156",
      windowID: "b03040dba1"
   }
 
ppxo_unhandled_error [object Object]
logger.js (65,13)
   "ppxo_unhandled_error"
   {
      [functions]: ,
      __proto__: { },
      country: "US",
      env: "production",
      errtype: "[object Error]",
      host: "jo-pp-node-demo.herokuapp.com",
      lang: "en",
      pageID: "030258b95a",
      path: "/",
      referer: "jo-pp-node-demo.herokuapp.com",
      stack: "Error: No window to focus
   at ParentComponent.prototype.focus (https://www.paypalobjects.com/api/checkout.js:5977:35)
   at options.overrides.focus (https://www.paypalobjects.com/api/checkout.js:5724:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3438:21)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3437:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2858:17)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at _RECEIVE_MESSAGE_TYPE[] (https://www.paypalobjects.com/api/checkout.js:2852:13)
   at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2785:25)
   at messageListener (https://www.paypalobjects.com/api/checkout.js:2806:13)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3384:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8202:37)
   at ZalgoPromise.prototype.dispatch (https://www.paypalobjects.com/api/checkout.js:8189:26)
   at ZalgoPromise.prototype.reject (https://www.paypalobjects.com/api/checkout.js:8176:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8214:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8202:37)
   at ZalgoPromise.prototype.dispatch (https://www.paypalobjects.com/api/checkout.js:8189:26)
   at ZalgoPromise.prototype.reject (https://www.paypalobjects.com/api/checkout.js:8176:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:8142:42)
   at responseListener.respond (https://www.paypalobjects.com/api/checkout.js:3639:43)",
      timestamp: 1509609617250,
      uid: "47c74f8417",
      ver: "4.0.156",
      windowID: "b03040dba1"
   }
 
SCRIPT5022: No window to focus
   at ParentComponent.prototype.focus (https://www.paypalobjects.com/api/checkout.js:5977:35)
   at options.overrides.focus (https://www.paypalobjects.com/api/checkout.js:5724:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3438:21)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3437:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2858:17)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at _RECEIVE_MESSAGE_TYPE[] (https://www.paypalobjects.com/api/checkout.js:2852:13)
   at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2785:25)
   at messageListener (https://www.paypalobjects.com/api/checkout.js:2806:13)
exceptions.js (14,9)
SCRIPT5022: No window to focus
   at ParentComponent.prototype.focus (https://www.paypalobjects.com/api/checkout.js:5977:35)
   at options.overrides.focus (https://www.paypalobjects.com/api/checkout.js:5724:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3438:21)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3437:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2858:17)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at _RECEIVE_MESSAGE_TYPE[] (https://www.paypalobjects.com/api/checkout.js:2852:13)
   at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2785:25)
   at messageListener (https://www.paypalobjects.com/api/checkout.js:2806:13)
exceptions.js (14,9)
SCRIPT5022: No window to focus
   at ParentComponent.prototype.focus (https://www.paypalobjects.com/api/checkout.js:5977:35)
   at options.overrides.focus (https://www.paypalobjects.com/api/checkout.js:5724:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3438:21)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3437:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2858:17)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at _RECEIVE_MESSAGE_TYPE[] (https://www.paypalobjects.com/api/checkout.js:2852:13)
   at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2785:25)
   at messageListener (https://www.paypalobjects.com/api/checkout.js:2806:13)
exceptions.js (14,9)
SCRIPT5022: No window to focus
   at ParentComponent.prototype.focus (https://www.paypalobjects.com/api/checkout.js:5977:35)
   at options.overrides.focus (https://www.paypalobjects.com/api/checkout.js:5724:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3438:21)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3437:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2858:17)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at _RECEIVE_MESSAGE_TYPE[] (https://www.paypalobjects.com/api/checkout.js:2852:13)
   at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2785:25)
   at messageListener (https://www.paypalobjects.com/api/checkout.js:2806:13)
exceptions.js (14,9)
SCRIPT5022: No window to focus
   at ParentComponent.prototype.focus (https://www.paypalobjects.com/api/checkout.js:5977:35)
   at options.overrides.focus (https://www.paypalobjects.com/api/checkout.js:5724:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3438:21)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3437:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2858:17)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at _RECEIVE_MESSAGE_TYPE[] (https://www.paypalobjects.com/api/checkout.js:2852:13)
   at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2785:25)
   at messageListener (https://www.paypalobjects.com/api/checkout.js:2806:13)
exceptions.js (14,9)
SCRIPT5022: No window to focus
   at ParentComponent.prototype.focus (https://www.paypalobjects.com/api/checkout.js:5977:35)
   at options.overrides.focus (https://www.paypalobjects.com/api/checkout.js:5724:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3438:21)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3437:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2858:17)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at _RECEIVE_MESSAGE_TYPE[] (https://www.paypalobjects.com/api/checkout.js:2852:13)
   at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2785:25)
   at messageListener (https://www.paypalobjects.com/api/checkout.js:2806:13)
exceptions.js (14,9)
SCRIPT5022: No window to focus
   at ParentComponent.prototype.focus (https://www.paypalobjects.com/api/checkout.js:5977:35)
   at options.overrides.focus (https://www.paypalobjects.com/api/checkout.js:5724:33)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3438:21)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:3437:17)
   at Anonymous function (https://www.paypalobjects.com/api/checkout.js:2858:17)
   at ZalgoPromise.try (https://www.paypalobjects.com/api/checkout.js:8325:21)
   at _RECEIVE_MESSAGE_TYPE[] (https://www.paypalobjects.com/api/checkout.js:2852:13)
   at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2785:25)
   at messageListener (https://www.paypalobjects.com/api/checkout.js:2806:13)
exceptions.js (14,9)

**Comment by bluepnume** [comment]
Hi @benzookapi, thanks for checking.

Most of the errors you shared show `Error: No window to focus` -- this seems to be a different error to the ones which were reported in the original issue.

Can you please give a description for your latest round of testing:

1. What were the specific steps you took?
2. What specifically went wrong with the experience? Did the window close? Or did something else happen?
3. Did the problem happen consistently, or intermittently?

**Comment by bluepnume** [comment]
Hi @benzookapi -- sorry to bother you but given our difficulty reproducing this problem, would really appreciate your help with the above questions. Thanks!

**Comment by benzookapi** [comment]
@bluepnume  Sorry for late reply.. I cannot spare the time to set up my own eivrionmnnt...

>What were the specific steps you took?

No but we tried couples of times.

>What specifically went wrong with the experience? Did the window close? Or did something else happen?

Not window closed, but the message like  "sorry, the payment cannot be proceeded" shown in the popup window.  (Not 100 %, 2 or 3 times out of 10 or more try)

>Did the problem happen consistently, or intermittently?

intermittently

>These Edge issues are really difficult to track down, 

I know that, but why I reported this is because one of my merchant  faced many issues in Edge (or IE) buyers and we need it stable.

>I've made some fixes for some Edge issues I found in the last few days. 

If you feel this difficult to be reproduced,   I can read the release history of this repo in GitHub and can detect what kind of modification has been given with release notes (comments) to determine whether Edge is stable successfully or not.  Am I correct?
